api/image_pull.go

```diff
@@ -16,11 +19,24 @@ func (c *API) ImagePull(ctx context.Context, nameWithTag string) error {
 	}
 
 	defer res.Body.Close()
-	body, _ := ioutil.ReadAll(res.Body)
-
 	if res.StatusCode != http.StatusOK {
+		body, _ := ioutil.ReadAll(res.Body)
 		return fmt.Errorf("unknown error, status code: %d: %s", res.StatusCode, body)
 	}
 
+	dec := json.NewDecoder(res.Body)
+	var report ImagePullReport
+	for {
+		if err = dec.Decode(&report); err == io.EOF {
+			break
+		} else if err != nil {
+			return fmt.Errorf("Error reading response: %v", err)
```

Comment:
@pablofsf please use %w for new code to wrap the error.

Reply:
Fixed!